### DISCUSSION TITLE
Interaction between proxy defined via `proxy` and `httpsAgent` settings

### DISCUSSION THREAD

**Discussion by danielsharvey** [axios/axios#5741] [General]
I had a situation with an interesting interaction between a proxy defined via the `proxy` config setting and a proxy defined by the `httpsAgent` config setting (using [https-proxy-agent](https://www.npmjs.com/package/https-proxy-agent)).

The `httpsAgent` was intentional.

The the `proxy` config setting was unintentional and coming from an `HTTPS_PROXY` environment variable setting.

With both defined, requests would always fails with the error:
```
Error: socket hang up
```

The solution was to specify  the `proxy: false` setting to "block" the effect of any environment variables.

My questions:

1. What should be expected in this case? i.e. will it always fail in this manner
2. Is there a way that this config "overlap" can be detected? (I suspect it would be hard, though perhaps at runtime?)

I'm interested in anyone elses experience.

**Comment by martines205** [comment]
did you have solve this?

**Comment by jasonsaayman** [comment]
`proxy: false` is the expected way to opt out of Axios' proxy handling, including proxy settings picked up from `HTTP_PROXY` / `HTTPS_PROXY`.

If you provide an `httpsAgent` that already handles proxying, Axios cannot reliably know that from the agent object. So if Axios also applies its own proxy config from the environment, you can effectively end up with overlapping proxy behavior, which commonly shows up as connection failures such as `socket hang up`.

Recommended pattern:

```js
import { HttpsProxyAgent } from 'https-proxy-agent';

axios.get(url, {
  httpsAgent: new HttpsProxyAgent(proxyUrl),
  proxy: false
});
```

So the short answer is: do not mix Axios `proxy` config/env proxy handling with a proxying `httpAgent`/`httpsAgent`; choose one path and disable the other with `proxy: false`.
